pyfr/plugins/tavg.py

```diff
@@ -86,15 +104,20 @@ def _init_gradients(self):
                            for pname in gradpnames]
 
     def _init_accumex(self, intg):
-        self.prevt = self.tout_last = intg.tcurr
+        comm, rank, root = get_comm_rank_root()
+        self.tstart_acc = self.prevt = self.tout_last = intg.tcurr
         self.prevex = self._eval_acc_exprs(intg)
         self.accex = [np.zeros_like(p, dtype=np.float64) for p in self.prevex]
-
-        # Extra state for continuous accumulation
-        if self.mode == 'continuous':
-            self.caccex = [np.zeros_like(a) for a in self.accex]
-            self.tstart_actual = intg.tcurr
-
+        self.vaccex = [np.zeros_like(a, dtype=np.float64) for a in self.accex]
+        self.sw = 0
+        self.W1m = 1
+        num = 0
+        for idx, etype, rgn in self._ele_regions:
+            soln = intg.soln[idx][..., rgn].swapaxes(0, 1)
```

Comment:
There are easier ways to get the relevant sizes. (See some of the attributes of the system and the `len(rgn)`.

Reply:
Would `intg.system.ele_shapes[idx][0]*len(rgn)` work?

Reply:
`ele_map[etype].nupts` is probably the best way to get the number of solution points.  Region size can be a bit more fiddly due to the '*' case.

Reply:
Don't we delete intg.system.ele_map?

Reply:
After initalisation.  So during init of the plugins it is available.